examples/tomo/astra_cpu_2d.py

```diff
@@ -0,0 +1,121 @@
+# Copyright 2014, 2015 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Examples using the ASTRA cpu backend."""
+
+# pylint: disable=invalid-name,no-name-in-module
+
+from __future__ import print_function, division, absolute_import
+import os.path as pth
+from future import standard_library
+
+standard_library.install_aliases()
+
+# External
+import numpy as np
+import matplotlib
+matplotlib.use('agg')
+import matplotlib.pyplot as plt
+
+# Internal
+from odl import (Interval, FunctionSpace, uniform_discr,
+                 uniform_discr_fromspace, uniform_sampling)
+from odl.tomo import Parallel2dGeometry, FanFlatGeometry
+from odl.tomo import (astra_cpu_forward_projector_call,
+                      astra_cpu_backward_projector_call)
+
+
+def save_slice(data, name):
+    """Save image.
+
+    Parameters
+    ----------
+    data : `DiscreteLp`
+    name : `str`
+    """
+    path = pth.join(pth.join(pth.dirname(pth.abspath(__file__)), 'data'),
+                    'astra')
+    filename = '{}.png'.format(name.replace(' ', '_'))
+    path = pth.join(path, filename)
+
+    data.show('imshow', saveto=path, title='{} [:,:]'.format(name))
+
+    plt.close()
+
+
+# Create `DiscreteLp` space for volume data
+nvoxels = (100, 110)
+discr_vol_space = uniform_discr([-1, -1.1], [1, 1.1], nvoxels,
+                                dtype='float32')
+# Create phantom data
```

Comment:
Saw that one (and considered trying it), but I'd personally prefer not to add dependencies (even optional ones) wherever avoidable. Even now we have issues of people not testing all code on the possible setups, if we start adding more we get an exponential growth of possibilities.

Anyway, the current code cannot be `@jit`'ed:

```python
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "D:\Anaconda\lib\site-packages\spyderlib\widgets\externalshell\sitecustomize.py", line 685, in runfile
    execfile(filename, namespace)
  File "D:\Anaconda\lib\site-packages\spyderlib\widgets\externalshell\sitecustomize.py", line 71, in execfile
    exec(compile(scripttext, filename, 'exec'), glob, loc)
  File "D:/GitHub/ODL/odl/util/phantom.py", line 416, in <module>
    shepp_logan_3d = shepp_logan(disc)
  File "D:/GitHub/ODL/odl/util/phantom.py", line 394, in shepp_logan
    return _phantom_3d(space, _modified_shepp_logan_ellipse_3d())
  File "D:\Anaconda\lib\site-packages\numba\dispatcher.py", line 172, in _compile_for_args
    return self.compile(sig)
  File "D:\Anaconda\lib\site-packages\numba\dispatcher.py", line 350, in compile
    flags=flags, locals=self.locals)
  File "D:\Anaconda\lib\site-packages\numba\compiler.py", line 644, in compile_extra
    return pipeline.compile_extra(func)
  File "D:\Anaconda\lib\site-packages\numba\compiler.py", line 359, in compile_extra
    raise e
NotImplementedError: offset=718 opcode=0x5e opname=LIST_APPEND
```

so I guess we'd have to rewrite it if we wanted that, we could try I guess, but anything less that a factor of at least 2 probably isn't worth the complications.